modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/js/EditAssetCategoryCPDisplayLayout.js

```diff
@@ -0,0 +1,201 @@
+		});
+	}
+
+	function createLabel(id, title) {
+		const labelContainer = document.createElement('span');
+
+		labelContainer.classList.add(
+			'label',
+			'label-dismissible',
+			'label-secondary'
+		);
+		labelContainer.setAttribute('id', id);
+
+		const linkContainer = document.createElement('span');
+
+		linkContainer.classList.add('label-item', 'label-item-expand');
+
+		const link = document.createElement('a');
+
+		link.setAttribute('href', '#' + id);
+		link.innerText = title;
+		linkContainer.appendChild(link);
+
+		const buttonContainer = document.createElement('span');
+
+		buttonContainer.classList.add('label-item', 'label-item-after');
+
+		const button = document.createElement('button');
+
+		button.classList.add('close');
+		button.setAttribute('aria-label', 'Close');
+		button.setAttribute('type', 'button');
+		button.addEventListener('click', handleCloseButtonClick);
+
+		const svg = document.createElementNS(
+			'http://www.w3.org/2000/svg',
+			'svg'
+		);
+
+		svg.classList.add('lexicon-icon', 'lexicon-icon-times');
+		svg.setAttribute('focusable', false);
+		svg.setAttribute('role', 'presentation');
+
+		const use = document.createElementNS(
+			'http://www.w3.org/2000/svg',
+			'use'
+		);
+
+		use.setAttribute(
+			'href',
+			`${Liferay.ThemeDisplay.getPathThemeImages()}/clay/icons.svg#times`
+		);
+
+		svg.appendChild(use);
+		button.appendChild(svg);
+		buttonContainer.appendChild(button);
+
+		labelContainer.appendChild(linkContainer);
+		labelContainer.appendChild(buttonContainer);
+
+		return labelContainer;
+	}
+
+	const categoriesContainer = document.getElementById(
+		`${portletNamespace}categoriesContainer`
+	);
+
+	function handleCloseButtonClick(event) {
+		const button = event.currentTarget;
+
+		categoriesContainer?.removeChild(button.parentElement.parentElement);
+	}
+
+	const classPK = document.getElementById(`${portletNamespace}classPK`);
+
+	const assetCategoryId = assetCategory?.getCategoryId();
+
+	if (assetCategoryId) {
+		const categoryNode = createLabel(
+			`category-${assetCategoryId}`,
+			assetCategory.getTitle(locale)
+		);
+
+		categoriesContainer?.appendChild(categoryNode);
+		classPK.value = assetCategoryId;
+	}
+
+	const selectCategoriesButton = document.getElementById(
+		`${portletNamespace}selectCategories`
+	);
+
+	selectCategoriesButton?.addEventListener('click', () => {
+		openSelectionModal({
+			buttonAddLabel: Liferay.Language.get('done'),
+			multiple: true,
+			onSelect: (selectedItems) => {
+				if (selectedItems) {
+					Object.keys(selectedItems).forEach((key) => {
+						const item = selectedItems[key];
+
+						if (item.unchecked && !categoriesContainer) {
+							return;
+						}
+
+						const categoryNodes = categoriesContainer.children;
+
+						if (categoryNodes.length) {
+							for (let i = 0; i < categoryNodes.length; i++) {
+								categoriesContainer.removeChild(
+									categoryNodes.item(i)
+								);
+							}
+						}
+
+						delete selectedItems.key;
```

Comment:
Hi @kresimir-coko, this was already there but is wrong - it's basically removing the same `key` property each single loop. Could you please 🙏  update it in `delete selectedItems[key];`

Reply:
![](https://media4.giphy.com/media/GCvktC0KFy9l6/giphy.gif)

Reply:
@FabioDiegoMastrorilli Fixed in https://github.com/liferay-commerce/liferay-portal/pull/1155/commits/ad5ef278751287411c1175832fdeafb5f1794d09

I've also rebased with `master`